### ISSUE TITLE
Only enable right sidebar on one view controller

### ISSUE THREAD

**Issue by amycheong19** [John-Lluch/SWRevealViewController#428]
I have a main left sidebar which it appear in most of the view controllers but my right sidebar only available for viewcontrollerA (one only). Is there a way for me to disable the right sidebar for other view controllers? it is because it appear when I swipe to delete my tableviewcell (pan gesture but pull out right sidebar suddenly)

**Comment by nazrdogan** [comment]
+1

**Comment by iDevelopper** [comment]
Yes, two solutions:

1) Set rightViewController to nil when you are in the others view controllers.
2) Implement the delegate method:

```
// Implement this to return NO when you want the pan gesture recognizer to be ignored
- (BOOL)revealControllerPanGestureShouldBegin:(SWRevealViewController *)revealController;
```

Check the pan direction: revealController.panGestureRecognizer.velocityInView. if < 0, pan direction is left and return false.

**Comment by nazrdogan** [comment]
I did like this 
        `self.revealViewController().rightViewController =  nil`

it works. but How can back to my controller
I also try
  `
 self.revealViewController().rightViewController =  MenuController
`
 not working.

**Comment by iDevelopper** [comment]
What is MenuController? Is it instantiate? If not instantiate it from your storyboard or when you first instantiate it in your application, save a reference to it.

**Comment by nazrdogan** [comment]
https://www.appcoda.com/sidebar-menu-swift/  I used this example . MenuController is a tableview Controller. but it uses segue. 
I did like this (I know its wrong)  but Menucontroller have two view. 

`
 let storyboard = UIStoryboard(name: "Main", bundle: nil)
 let vc = storyboard.instantiateViewController(withIdentifier: "MenuController") as! UIViewController
self.revealViewController().rightViewController =  vc
`

**Comment by iDevelopper** [comment]
And what is the problem? Could you explain more? What's happen?

**Comment by nazrdogan** [comment]
ok .I solved  .in this example right and rear menu view has no storyboard id. I give  rightMenu id for right menu storyboard view.  and 

<pre>
  let storyboard = UIStoryboard(name: "Main", bundle: nil)
  let vc = storyboard.instantiateViewController(withIdentifier: "rightMenu")
  self.revealViewController().rightViewController =  vc
</pre>

now its working. but is it right way ?

**Comment by iDevelopper** [comment]
Yes. You could also keep a reference of this controller in a variable (when you instantiate it for the first time), if you don't want to reload this controller each time.

**Comment by nazrdogan** [comment]
@iDevelopper  Thanks :) . I will do it.

**Comment by iDevelopper** [comment]
Hi @nazrdogan , this is a sample:

[SWRight.zip](https://github.com/John-Lluch/SWRevealViewController/files/502277/SWRight.zip)

**Comment by nazrdogan** [comment]
its awesome example :) . Thank you again and again 👍
